lib/facility_factory.rb

```diff
@@ -0,0 +1,50 @@
+class FacilityFactory
+  attr_reader :facilities
+
+  def initialize
+    @facilities = []
+  end
+
+  def create_facilities(dmv_locations_data)
+    dmv_locations_data.each do |dmv|
+      name = dmv[:dmv_office] || dmv[:office_name] || dmv[:name]
+      phone = dmv[:phone] || dmv[:public_phone_number]
+      services = dmv[:services_p]
+      address_line_1 = dmv[:address_li] || 
+                      dmv[:street_address_line_1] ||
+                      dmv[:address1]
+      address_line_2 = dmv[:address__1] || 
+                      dmv[:street_address_line_2]
+      address_line_3 = dmv[:location]
+      city = dmv[:city]
+      state = dmv[:state]
+      zip = dmv[:zip] || 
+            dmv[:zip_code] || 
+            dmv[:zipcode]
+      if address_line_3 != nil
+        address = address_line_1 + " " + 
+                  address_line_2 + " " +
+                  address_line_3 + " " +
+                  city + " " +
+                  state + " " +
+                  zip
+      elsif address_line_2 == nil && address_line_3 == nil
+        address = address_line_1 + " " + 
+                  city + " " +
+                  state + " " +
+                  zip
+      else
+        address = address_line_1 + " " + 
+                  address_line_2 + " " +
+                  city + " " +
+                  state + " " +
+                  zip
+      end
+      @facilities << Facility.new({name: name,
+                                  address: address, 
+                                  phone: phone,
+                                  services: services})
+    end
+  end
```

Comment:
I like that you accounted for all these scenarios where address lines are present or not and you've also made it so this one method can take in any of the state's facilities. I do however, think this method is doing too much and could benefit from being refactored. Pulling out pieces of logic into helper methods will help this be more readable and easier to navigate and debug.